### PR TITLE
Added reference for boilerplate text.

### PR DESCRIPTION
**Pull request by rbeyer**
## Description
Addition of a README file and a *very minimal* default.cpp file in the existing (but empty) isis/templates/ directory in order to provide a reference for developers about what the definitive boilerplate text is that should be at the top of *all* files, also establishes this as a place to locate starter-files that contain best practices of includes or layouts or whatever.

## Related Issue
#3741 

## Motivation and Context
Now that the new LICENSE file is in place, the discussion in #3741 also needed a resolution in the repo about *exactly* what the boilerplate text is.  This provides that resolution.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/) -- **can't still 404ing**.
- [X] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/templates/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+is placed in a comment near the top of the file, or in some other
+appropriate location, depending on file type.
+
+    This is free and unencumbered software released into the public domain.
+
+    The authors of ISIS do not claim copyright on the contents of this file.
+    For more details about the LICENSE terms and the AUTHORS, you will
+    find files of those names at the top level of this repository.
+
+Also include this as a separate comment underneath the above,
```
Comment (jessemapel) [inline isis/templates/README.md]: Do you mean separate line?

File: isis/templates/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+to make it easier to find-and-replace if we ever need to:
+
+    SPDX-License-Identifier: CC0-1.0
+
+Configuration files and minor READMEs like this one don't really
+need it, as the blanket LICENSE file covers those to the extent
+needed.  PR reviewers should be able to help advise if you are
```
Comment (jessemapel) [inline isis/templates/README.md]: I'd rather this sentence read "Add a comment on your PR if you are unsure if a file needs these comments." It's not clear what contributors may be unsure of. Also, it's a bit more actionable.

**Review by jessemapel — COMMENTED** [review]
This looks good. Once this is merged, I may put in a header template file too.

File: isis/templates/README.md (hunk 2 above)
Comment (rbeyer) [reply isis/templates/README.md]: Good point, will include in commit when you help me resolve the above.

**Review by jlaura — DISMISSED** [review]
Approving. Thanks for putting in the PR @rbeyer. Once the comments from @jessemapel are resolved, I think this is good to merge.

File: isis/templates/README.md (hunk 1 above)
Comment (rbeyer) [reply isis/templates/README.md]: Whoops, sorry, I could have sworn I wrote something here, I even referenced it in the next comment down.

In Jay's C++ example, he had the SPDX line as a separate comment, that's how I implemented it in default.cpp, and that's what 'separate comment' was referring to.  Maybe we don't care, and we just have it all in one comment block, with the SPDX line separate in the example, and folks will just 'do the right thing' when they implement it, helped by specific examples?  I have no strong feelings on this, and am happy to take direction.

Comment (rbeyer) [reply isis/templates/README.md]: I just made the change to remove that weird distinction, examples are better, anyway.

**Review by jessemapel — APPROVED** [review]

**Pull request merged**